### PR TITLE
gh-104804: Remove webbrowser.MacOSX class, deprecated in Python 3.11

### LINKED ISSUE #104804: Remove deprecated webbrowser.MacOSX class
# Feature or enhancement

The `webbrowser.MacOSX` class was deprecated in Python 3.11 and scheduled for removal in 3.13.

Let's remove it. PR to follow.

# Previous discussion

* GH-86421

### Linked PRs
* gh-104816

### PR DESCRIPTION
**Pull request by hugovk**
Closes #104804.

Remove `webbrowser.MacOSX class`, deprecated in Python 3.11 in gh-86421.

---

Also fix Sphinx warnings in `webbrowser.rst` (helps gh-101100):

* I don't think we need to define `envvar`s for `PATH` or `KDEDIR`

* We could define `.. envvar:: BROWSER` right before the first mention, but it looks a little out of place having a `BROWSER` before the third paragraph. Thoughts?

* I also don't think we need to define the dozen or so `Mozilla`, `Opera` etc. classes. They're listed in the table that shows which type name corresponds to each class, and some of them have extra details like how they're instantiated (`Mozilla('mozilla')`, `GenericBrowser('lynx')`), so plain literal text makes sense here. Repeating each class below would be a lot of duplication for little benefit, especially for this rather short page.

* https://cpython-previews--104816.org.readthedocs.build/en/104816/library/webbrowser.html

----
:books: Documentation preview :books:: https://cpython-previews--104816.org.readthedocs.build/

* Issue: gh-104804

### CODE REVIEW

**Comment by serhiy-storchaka** [comment]
Would not be better to split this PR on two? One for removing, and other for tidying?

**Comment by hugovk** [comment]
I can do, but they are intertwined: this bit is removed, which otherwise has warnings that would need fixing, or waiting for part 1 to be merged:

```rst
.. deprecated-removed:: 3.11 3.13
   :class:`MacOSX` is deprecated, use :class:`MacOSXOSAScript` instead.
```

Seems easier to do both at once. But either way is fine for me :)

File: Misc/NEWS.d/next/Library/2023-05-23-21-25-54.gh-issue-104804.78fiE6.rst
Comment (CAM-Gerlach) [inline Misc/NEWS.d/next/Library/2023-05-23-21-25-54.gh-issue-104804.78fiE6.rst]:
```suggestion
Remove the :module:`webbrowser` :class:`!webbrowser` class, deprecated in Python 3.11.
```

* Link to the module docs
* Use similar phrasing to other changelog entries, framed as complete sentances

File: Misc/NEWS.d/3.11.0a1.rst
Comment (CAM-Gerlach) [inline Misc/NEWS.d/3.11.0a1.rst]:
```suggestion
It is untested and undocumented and also not used by :mod:`webbrowser` itself.
```

Link to the module, now that the above doesn't.

File: Doc/whatsnew/3.13.rst
Comment (CAM-Gerlach) [inline Doc/whatsnew/3.13.rst]:
```suggestion
* Remove the untested and undocumented :mod`webbrowser` :class:`!MacOSX` class,
  deprecated in Python 3.11.
  Use the :class:`!MacOSXOSAScript` class (introduced in Python 3.2) instead.
```

* Briefly justify the change
* Mention its replacement
* Link to the module for reference
* Use the same "the"/complete sentence phrasing as the others

File: Doc/library/webbrowser.rst
Comment (CAM-Gerlach) [inline Doc/library/webbrowser.rst:176]: Can't reach it, but as previously discussed I would suggest adding the missing `.. versionchanged:: 3.2` above the 3.3 that `MacOSXOSAScript` was added and used by default instead of `MacOSX` for Mac. This retains the remaining relevant bit from the `deprecated-removed` message, in a more useful fashion for readers.

E.g. something like

```rst
.. versionadded:: 3.2
   A new :class:`!MacOSXOSAScript` class has been added
   and is used on Mac instead of the previous :class:`!MacOSX` class.
   This adds support for opening browsers not currently set as the OS default.
```

**Review by CAM-Gerlach — COMMENTED** [review]
I had a handful of doc-related suggestions. Otherwise, the core change looks good. The removal of the `:class:` syntax for things that are really code snippits (and not formally documented anyway) makes sense, as does silencing warnings for env vars not explicitly defined by or primarily/exclusively used by Python itself.

Ideally it would be nice to split it into two PRs as @serhiy-storchaka , but as a practical matter given they interlinked, maybe not worth the extra effort.

File: Doc/library/webbrowser.rst
Comment (hugovk) [reply Doc/library/webbrowser.rst:176]: Thanks, updated!

File: Doc/library/webbrowser.rst
Comment (vstinner) [inline Doc/library/webbrowser.rst:178]: While the MacOSX class was not explicitly documented, it was mentioned in the deprecation notice. What do you think about documenting its removal here?

File: Misc/NEWS.d/next/Library/2023-05-23-21-25-54.gh-issue-104804.78fiE6.rst
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2023-05-23-21-25-54.gh-issue-104804.78fiE6.rst]:
```suggestion
Remove the untested and undocumented :mod:`webbrowser` :class:`!MacOSX` class, deprecated in Python 3.11.
```

**Review by vstinner — APPROVED** [review]
LGTM if you fix the typo in the NEWS entry.

File: Doc/library/webbrowser.rst
Comment (hugovk) [reply Doc/library/webbrowser.rst:178]: I considered it but wasn't sure, can definitely add if you like?

**Pull request merged**
